test/external_tests.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+from argparse import ArgumentParser, Namespace
+import os
+from os import listdir
+from pathlib import Path
+import sys
+
+# Our scripts/ is not a proper Python package so we need to modify PYTHONPATH to import from it
+# pragma pylint: disable=import-error,wrong-import-position
+SCRIPTS_DIR = Path(__file__).parents[1] / "scripts"
+sys.path.insert(0, str(SCRIPTS_DIR))
+from common.shell_command import run_cmd
+
+EXTERNAL_TESTS_DIR = Path(__file__).parent / "externalTests"
+
+
+class OptionNotFound(Exception):
+    pass
+
+
+class ExternalTestNotFound(Exception):
+    pass
+
+
+def external_tests_scripts() -> dict:
+    test_scripts: dict = {}
+    for f in listdir(EXTERNAL_TESTS_DIR):
+        file_path = Path(EXTERNAL_TESTS_DIR) / f
+        if file_path.is_file():
+            file_name, extension = file_path.name.split(".")
+            # TODO: Remove `file_name == "common"` when we complete the migration
+            # of the external tests to python, since there will be no more
+            # common.sh script in the externalTests folder.
+            if extension == "sh" and not file_name == "common":
+                test_scripts[file_name] = file_path
+    return test_scripts
+
+
+def display_available_external_tests(_) -> int:
+    print("Available external tests:")
+    print(*external_tests_scripts().keys())
+    return os.EX_OK
+
+
+def run_test_script(solc_binary_type: str, solc_binary_path: Path, tests: dict) -> int:
+    for test_name, test_script_path in tests.items():
+        print(f"Running {test_name} external test...")
+        ret = run_cmd(f"{test_script_path} {solc_binary_type} {solc_binary_path}")
+        if ret != os.EX_OK:
+            return ret
```

Comment:
You're using `check=True` so you'll get an exception if the code is not zero:
```suggestion
        ret = run_cmd(f"{test_script_path} {solc_binary_type} {solc_binary_path}")
        assert ret == os.EX_OK
```

Reply:
Actually, I just realized that I should have suggested putting the `assert` in `run_cmd()` so that you don't have to assert this every time you use it.

But I guess this will solve itself if you remove the wrapper.